rama-net/src/client/pool.rs

```diff
@@ -131,13 +132,59 @@ struct PooledConnection<C, ID> {
     last_used: Instant,
 }
 
+pub type FiFoReuseLruDropPool<C, ID> = LruDropPool<FiFoReuse, C, ID>;
+pub type RoundRobinReuseLruDropPool<C, ID> = LruDropPool<RoundRobinReuse, C, ID>;
+
 /// Connection pool that uses FiFo for reuse and LRU to evict connections
-pub struct FiFoReuseLruDropPool<C, ID> {
+pub struct LruDropPool<R, C, ID> {
     storage: Arc<Mutex<VecDeque<PooledConnection<C, ID>>>>,
     total_slots: Arc<Semaphore>,
     active_slots: Arc<Semaphore>,
     idle_timeout: Option<Duration>,
     returner: ConnReturner<C, ID>,
+    _marker: PhantomData<R>,
+}
+
+pub struct FiFoReuse;
+pub struct RoundRobinReuse;
+
+trait Reuse<C, ID> {
+    fn get_conn(
+        storage: &mut VecDeque<PooledConnection<C, ID>>,
+        id: &ID,
+    ) -> Option<(usize, PooledConnection<C, ID>)>;
+}
+
+impl<C, ID> Reuse<C, ID> for FiFoReuse
```

Comment:
Am I going crazy? Do you not just do exactly the same in the `Reuse` impl of `FiFoReuse` and `RoundRobinReuse`?

```
let idx = storage.iter().position(|stored| &stored.id == id)?;
```

```
let idx = storage
            .iter()
            .enumerate()
            .find(|(_, stored)| &stored.id == id)?
            .0;
 ```
 
 These are equivalent, no??